packages/teuchos/core/src/Teuchos_ScalarTraits.cpp

```diff
@@ -119,7 +120,7 @@ bool Teuchos::operator&&(const qd_real &a, const qd_real &b) {
 #ifndef __sun
```

Comment:
I'm pretty sure we could purge the work-arounds for broken old Sun compilers, but we don't have to do that in this commit.

Reply:
There's several of them.

```
dir-teuchos (gcc) 261 $ find . -name \*pp -exec grep -l __sun {} \;
./core/test/MemoryManagement/ArrayRCP_test.cpp
./core/src/Teuchos_ArrayView.hpp
./core/src/Teuchos_ScalarTraits.hpp
./core/src/Teuchos_ArrayViewDecl.hpp
./core/src/Teuchos_ScalarTraits.cpp
./remainder/src/Teuchos_FloatingPointTrap.cpp

```